packages/loopring_v3/contracts/core/impl/libexchange/ExchangeDeposits.sol

```diff
@@ -78,22 +75,17 @@ library ExchangeDeposits
         bytes   memory extraData
         )
         private
-        returns (
-            uint96 amountDeposited,
-            uint64 fee
-        )
+        returns (uint96 amountDeposited)
     {
         IDepositContract depositContract = S.depositContract;
-        uint depositValueETH = 0;
-        if (msg.value > 0 && (tokenAddress == address(0) || depositContract.isETH(tokenAddress))) {
-            depositValueETH = amount;
-            fee = uint64(msg.value.sub(amount));
+        if (tokenAddress == address(0) || depositContract.isETH(tokenAddress)) {
+            require(msg.value == amount, "INVALID_AMOUNT");
         } else {
-            fee = uint64(msg.value);
+            require(msg.value == 0, "INVALID_AMOUNT");
         }
```

Comment:
I think this can now be removed entirely, the deposit contract can and does check this as well.